### ISSUE TITLE
Cookies being blocked in embedded app

### ISSUE THREAD

**Issue by andersonjoseph** [Shopify/shopify-app-template-remix#585]
# Issue summary

- `@shopify/shopify-app-remix` version: 2.1.0
- Node version: 20.2.0
- Operating system: Linux Mint 21.3 x86_64

I'm developing an embedded app using `createCookieSessionStorage` to store tokens and other data related to user session. However it does not work for browsers that are blocking 3rd party cookies. According to [this comment](https://github.com/Shopify/shopify-app-template-remix/issues/552#:~:text=Cookie%2Dbased%20session%20authentication%20is%20not%20recommended%20for%20embedded%20apps%20anymore) Cookie-based session authentication is not recommended for embedded apps anymore.

What could be a good alternative for cookies in embedded apps? I tried `localstorage` but it's not available in server-side functions (`loader` or `action`)

**Comment by paulomarg** [comment]
Hey! Unfortunately (or fortunately!) browsers are making it more difficult to use cookies within iframes (i.e. 3rd party cookies), because they pose a privacy risk for users, as they can be used for tracking user activity. The same goes for localStorage and the likes.

A common alternative is to use a `Bearer` token in the `Authorization` header using a shared private key - this is how app bridge authenticates requests to the backend, and what `authenticate.admin` will look for. The remix package will then store the session information using your configured `sessionStorage` (in `shopify.server.js`).

Do you need to store any specific information in the session, in addition to [the defaults we store](https://github.com/Shopify/shopify-app-template-remix/blob/main/prisma/schema.prisma#L16)?

**Comment by andersonjoseph** [comment]
Yes, I ended up using an approach similar to the one described here: ["How Can I Save Multiple Remix Based Custom App Auth Session In Same Database"](https://community.shopify.com/c/authentication-and-access/how-can-i-save-multiple-remix-based-custom-app-auth-session-in/m-p/2379799/highlight/true#M1974)

```
model Session {
  id              String    @id
  shop            String
  state           String
  isOnline        Boolean   @Default(false)
  scope           String?
  expires         DateTime?
  accessToken     String
  userId          BigInt?
}

model MyAppSession {
  id           String     @id
  someData     String
}
```

I used drizzle instead of prisma but the idea is the same, right now is working effectively

**Comment by paulomarg** [comment]
Yeah, that is the approach I'd take in this case too. Since it seems like you're unblocked, I'm going to close this issue, but let us know if you have other problems!
